### PR TITLE
add "Learn" page

### PR DESCRIPTION
**Pull request by jchavarri**
Based on #1 (`landing-and-nav-structure`).

Adds a first version of the "Learn" page. It the content for the first two sections:
- New to OCaml?
- Why Melange

And provides a skeleton structure for the rest of the sections:
- Build system and package management
- Communicate with JavaScript
- Additions to OCaml
- Melange for X users

### CODE REVIEW

File: docs/learn.md [hunk 1]
```diff
@@ -1,3 +1,153 @@
...
+throughout these docs.
+
+If you are completely new to OCaml, you might want to get familiar first with
+the language, there are [plenty of resources available](https://ocaml.org/docs),
+but we recommend the following tutorials from the official OCaml website:
+
+- [A First Hour With OCaml](https://ocaml.org/docs/first-hour)
+- [OCaml Exercises](https://ocaml.org/problems)
+
+## Why Melange
```
Comment (anmonteiro) [inline docs/learn.md:17]: I think this and the history paragraphs don't belong in the "Learn" page. IMO this page should be actionable and straight to the point.

We should make a separate page for "rationale" / "history".

File: docs/learn.md [hunk 2]
```diff
@@ -1,3 +1,153 @@
...
+
+To achieve this goal, Js_of_ocaml picks the bytecode generated by OCaml [batch
+compilation](https://ocaml.org/manual/comp.html) and generates JavaScript from
+it. OCaml bytecode has a very stable interface, so Js_of_ocaml can easily
+upgrade to new versions of the compiler. Due to this design decision, it can
+also remain compatible with most of the OCaml ecosystem, as long as the tools or
+libraries don't rely on C code.
+
+The downside of using bytecode is that it gets harder to communicate with
+existing JavaScript code. For example, a Js_of_ocaml string is not compiled as a
```
Comment (anmonteiro) [inline docs/learn.md:57]: Didn't they recently make strings compatible with the JS side?

File: docs/learn.md [hunk 3]
```diff
@@ -1,3 +1,153 @@
...
+JavaScript code through a rich collection of attributes applied to `external`
+primitives.
+
+#### Reason
+
+Around the same year, a project called [Reason](https://reasonml.github.io/)
+appears at Facebook. Led by Jordan Walke, the idea is to create an alternate
+syntax for OCaml that is closer to C and JavaScript. Even if Reason has no take
+on which platform the code is deployed —native applications binaries, or web
+applications using JavaScript as a target language—, BuckleScript adds first
```
Comment (anmonteiro) [inline docs/learn.md:86]: this needs a space after the emdash.

File: docs/learn.md [hunk 4]
```diff
@@ -1,3 +1,153 @@
...
+to ReScript, António Monteiro starts working on a fork of BuckleScript with a
+simple (not easy) goal: replace the [Ninja build
+system](https://ninja-build.org/), which BuckleScript had been using from its
+creation, with [Dune](https://dune.build/), which is the most used build system
+for OCaml projects.
+
+This fork of BuckleScript is later named Melange. After finishing the switch
+from Ninja to Dune, several additional features were added to bring it closer to
+OCaml. Some examples are the upgrade of the compiler version used by Melange to
+4.12, or modelling the changes to the OCaml compiler that Melange uses as just a
```
Comment (anmonteiro) [inline docs/learn.md:126]: 4.12 and, later, to 4.14.

File: docs/learn.md [hunk 5]
```diff
@@ -1,3 +1,153 @@
...
+plain library, instead of a full fork of the whole compiler.
```
Comment (anmonteiro) [inline docs/learn.md:127]: instead of "whole" compiler, let's say "upstream compiler" or "stock OCaml compiler".

File: docs/learn.md (hunk 1 above)
Comment (jchavarri) [reply docs/learn.md:17]: You suggested the other day to have all this content in a single page. The more content we add, the more I like this approach :) It allows to keep the navigation pretty minimal (Get started, Learn, API, Try, Community), so I am a bit unconvinced that history/rationale has enough weight to deserve its own top level section.

Would it be better if we move the Why & History to the bottom of the "Learn" page? That way, when users reach the page, they can go straight to the point as you say, but we aren't forced to add a new top-level section just for that.

Comment (anmonteiro) [reply docs/learn.md:17]: my suggestion was about navigation, i.e. make it possible to have a single page with all the text on the website, but that's not strictly necessary.

I'd still prefer if we keep logically separate docs in separate files. `mkdocs` doesn't allow to have a single page that concats all pages?

Comment (jchavarri) [reply docs/learn.md:17]: > `mkdocs` doesn't allow to have a single page that concats all pages?

There is [some plugin](https://github.com/timvink/mkdocs-print-site-plugin) we could try.

> my suggestion was about navigation, i.e. make it possible to have a single page with all the text on the website, but that's not strictly necessary.
> I'd still prefer if we keep logically separate docs in separate files.

I think I misunderstood the whole thing, I can restructure the docs in separate files then. To clarify, the top sections would be:

- Get started
- Rationale <- new
- Learn
- API
- Try
- Community

?

File: docs/learn.md (hunk 2 above)
Comment (jchavarri) [reply docs/learn.md:57]: Right, I fixed this text in https://github.com/melange-re/documentation-site/commit/6cedebd48d2462bca74b4cee8d18952bd0d839d1.

File: docs/learn.md (hunk 1 above)
Comment (jchavarri) [reply docs/learn.md:17]: Moved in https://github.com/melange-re/documentation-site/commit/6cedebd48d2462bca74b4cee8d18952bd0d839d1.

File: docs/learn.md (hunk 3 above)
Comment (jchavarri) [reply docs/learn.md:86]: You mean space on both sides around em dashes? I thought that was [optional](https://www.merriam-webster.com/words-at-play/em-dash-en-dash-how-to-use):

> Spacing around an em dash varies. Most newspapers insert a space before and after the dash, and many popular magazines do the same, but most books and journals omit spacing, closing whatever comes before and after the em dash right up next to it. This website prefers the latter, its style requiring the closely held em dash in running text.

File: docs/learn.md (hunk 4 above)
Comment (jchavarri) [reply docs/learn.md:126]: It seems I changed this at some point after the review, now it says:

> Some examples are the upgrade of the OCaml compiler version used by Melange, or modeling the changes...

File: docs/learn.md (hunk 5 above)
Comment (jchavarri) [reply docs/learn.md:127]: https://github.com/melange-re/documentation-site/commit/02b279f79294ad3fcf7295593691a27a140523ad

File: docs/learn.md [hunk 6]
```diff
@@ -1,3 +1,153 @@
...
-_TODO: This will be a one-page with "all the things" in it_
+## New to OCaml?
+
+As Melange is a fork of the OCaml compiler, this documentation is designed to
+include the differences between both. But for the parts that overlap between
+Melange and OCaml, we will always refer to the main OCaml documentation
+throughout these docs.
+
+If you are completely new to OCaml, you might want to get familiar first with
+the language, there are [plenty of resources available](https://ocaml.org/docs),
```
[the remark this answers is not in the document]
Comment (jchavarri) [reply docs/learn.md:11]: https://github.com/melange-re/documentation-site/commit/02b279f79294ad3fcf7295593691a27a140523ad

File: docs/learn.md (hunk 3 above)
Comment (anmonteiro) [reply docs/learn.md:86]: Yeah I've seen you use this style in other places too. No strong opinions, then.

**Pull request merged**